src/test/ui/associated-item/associated-item-type-issue-63594.rs

```diff
@@ -0,0 +1,22 @@
+#![feature(associated_type_bounds)]
```

Comment:
```suggestion
// This test documents that `type Out = Box<dyn Bar<Assoc: Copy>>;`
// is allowed and will correctly reject an opaque `type Out` which
// does not satisfy the bound `<TheType as Bar>::Assoc: Copy`.
//
// FIXME(rust-lang/lang): I think this behavior is logical if we want to allow
// `dyn Trait<Assoc: Bound>` but we should decide if we want that. // Centril
//
// Additionally, as reported in https://github.com/rust-lang/rust/issues/63594,
// we check that the spans for the error message are sane here.

#![feature(associated_type_bounds)]
```

Also please move this file to `src/test/ui/associated-type-bounds/assoc-type-eq-with-dyn-atb-fail.rs` instead.

I already have this test in one of my branches (with that extra comment ^--) with the wrong spans but correct spans is not the main thing the tests is meant to test. ;)